staging/src/k8s.io/apiserver/pkg/authentication/request/x509/x509.go

```diff
@@ -276,10 +277,21 @@ var CommonNameUserConversion = UserConversionFunc(func(chain []*x509.Certificate
 	if len(chain[0].Subject.CommonName) == 0 {
 		return nil, false, nil
 	}
-	return &authenticator.Response{
-		User: &user.DefaultInfo{
-			Name:   chain[0].Subject.CommonName,
-			Groups: chain[0].Subject.Organization,
+
+	// The credential ID is the signature of the certificate.  This will be
+	// globally unique; if two certificates share the same signature then they
+	// must be the same certificate with extremely high probability.
+	credentialID := "X509Sig=" + base64.StdEncoding.EncodeToString(chain[0].Signature)
+
+	u := &user.DefaultInfo{
```

Comment:
Done.

Reply:
Actually, it is not possible to totally inline when using sha256, since the hash is returned as a fixed-length array.  To convert to a slice, I have to assign it to a variable.